WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/BuiltInCardReaderPaymentAlertsProvider.swift

```diff
@@ -42,20 +42,19 @@ final class BuiltInCardReaderPaymentAlertsProvider: CardReaderTransactionAlertsP
     }
 
     func success(printReceipt: @escaping () -> Void,
-                 emailReceipt: @escaping () -> Void,
-                 noReceiptAction: @escaping () -> Void,
-                 email: String?) -> CardPresentPaymentsModalViewModel {
-        if let email = email, email.isNotEmpty {
-            return CardPresentModalBuiltInSuccessEmailSent(printReceipt: printReceipt,
-                                                           noReceiptAction: noReceiptAction,
-                                                           email: email)
-        } else if MFMailComposeViewController.canSendMail() {
-            return CardPresentModalBuiltInSuccess(printReceipt: printReceipt,
-                                                  emailReceipt: emailReceipt,
-                                                  noReceiptAction: noReceiptAction)
-        } else {
-            return CardPresentModalBuiltInSuccessWithoutEmail(printReceipt: printReceipt,
-                                                              noReceiptAction: noReceiptAction)
+                 emailReceipt: CardReaderTransactionAlertEmailReceiptAction,
+                 noReceiptAction: @escaping () -> Void) -> CardPresentPaymentsModalViewModel {
+        switch emailReceipt {
+        case let .emailSent(email):
+            return CardPresentModalSuccessEmailSent(printReceipt: printReceipt,
+                                                    noReceiptAction: noReceiptAction,
+                                                    email: email)
+        case let .sendEmail(emailReceipt):
+            return CardPresentModalSuccess(printReceipt: printReceipt,
+                                           emailReceipt: emailReceipt,
+                                           noReceiptAction: noReceiptAction)
```

Comment:
I found these two hard to read when they're together as are very similar. How about changing the `emailReceipt` property to something like `emailReceiptAction` for consistency with `noReceiptAction`, and rename both `.emailSent` and `.sendEmail` cases to something more descriptive with what they'll be presenting? like promptToSendEmailReceipt, paymentSuccessEmailSent, or similar 🤔

Reply:
Good idea. I went back-and-forth with various versions of these. The goal of it all is clarity and if it's not clear it needs to be improved.